superset/connectors/sqla/models.py

```diff
@@ -977,6 +977,17 @@ def _get_top_groups(
         return or_(*groups)
 
     def query(self, query_obj: Dict) -> QueryResult:
+        filters = security_manager.get_row_level_security_filters(
```

Comment:
This new block should probably be in it's own method `_get_row_level_filters`

Reply:
Also wondering if this logic should be in `get_sqla_query` instead (unclear). One thing that would be neat would be to enable templating in the text, and if this was done in `get_sqla_query` you'd have a handle on `template_processor.process_template` in that context. 

Eventually it'd be great to be able to base row level rules on user attribute as in `WHERE sales_team_id = {{ current_user.extras.sales_team_id }}`, (note that `extras` doesn't really exist yet) instead of having to create entries for each sales team. You could even imagine things like `WHERE country IN {{ call_external_service.list_access_to_countries(current_user.id) }}`

Reply:
Moved to within `get_sqla_query`, using `text`.  I'd like the user attributes very much, eventually.